### PR TITLE
Add option to configure the API call lead time for timeout extensions

### PR DESCRIPTION
**Pull request by TomFrost**

### CODE REVIEW

**Comment by Fluidbyte** [comment]
LGTM

File: src/TimeoutExtender.js [hunk 1]
```diff
@@ -60,6 +65,7 @@ class TimeoutExtender {
     this._squiss.on('message', msg => this.addMessage(msg))
     this._visTimeout = this._opts.visibilityTimeoutSecs * 1000
     this._stopAfter = Math.min(this._opts.noExtensionsAfterSecs * 1000, MAX_MESSAGE_AGE_MS)
+    this._apiLeadMs = Math.min(this._opts.advancedCallMs, this._visTimeout * 1000)
```
Comment (stephen-ta) [inline src/TimeoutExtender.js]: `this._visTimeout` was already multiplied by `1000`

**Comment by stephen-ta** [comment]
One small change may be necessary, but 🍪  after that!

File: src/TimeoutExtender.js (hunk 1 above)
Comment (TomFrost) [inline src/TimeoutExtender.js]: Gah, good catch! I had this as opts.visibilityTimeoutSecs (hence the `* 1000`) but changed it to the one that was already multiplied after the fact. This could probably use an edge case test, since everything still passed...

Comment (TomFrost) [inline src/TimeoutExtender.js]: Fixed

**Comment by TomFrost** [comment]
(Fail was a connectivity issue)
